crates/ignore/src/types.rs

```diff
@@ -277,31 +485,57 @@ impl Types {
         let name = match file_name(path.as_ref()) {
             Some(name) => name,
             None if self.has_selected => {
-                return Match::Ignore(Glob::unmatched());
+                return Match::Ignore(Globs::unmatched());
             }
             None => {
                 return Match::None;
             }
         };
         let mut matches = self.matches.get_or_default().borrow_mut();
         self.set.matches_into(name, &mut *matches);
-        // The highest precedent match is the last one.
-        if let Some(&i) = matches.last() {
+        let mut file_type_indices_token_and_is_negated: Option<(
+            FileTypeIndicesToken,
+            bool,
+        )> = Default::default();
+        for &i in &*matches {
             let (isel, _) = self.glob_to_selection[i];
             let sel = &self.selections[isel];
-            let glob = Glob(GlobInner::Matched { def: sel.inner() });
```

Comment:
This used to "expose" (but only internally I think?) a single matching `FileTypeDef`, but we want _all_ of them (for our `tree-sitter-grep` use case)